### PR TITLE
openssh-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
SLES:
JOB ID     : 098deaf2246433cb4de76437da8c498494a2984b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T12.58-098deaf/job.log
 (1/2) openssh-tests.py:OpenSSH.test;run-run_type-upstream-7115: FAIL: openssh-tests.py: 'make check' failed. (25.13 s)
 (2/2) openssh-tests.py:OpenSSH.test;run-run_type-distro-f473: ERROR: openssh source install failed. (3.18 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T12.58-098deaf/results.html
JOB TIME   : 46.86 s

RHEL:
JOB ID     : c727607d644422b70cc81843dff4f8f78b964758
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-27T02.33-c727607/job.log
 (1/2) openssh-tests.py:OpenSSH.test;run-run_type-upstream-7115: FAIL: openssh-tests.py: 'make check' failed. (411.32 s)
 (2/2) openssh-tests.py:OpenSSH.test;run-run_type-distro-f473: FAIL: openssh-tests.py: 'make check' failed. (99.05 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-27T02.33-c727607/results.html
JOB TIME   : 521.32 s
```

**Comment by nasastry** [comment]
```
SLES:
JOB ID     : cafb83334f1d0147baf9a326a7f5e16ec34ac2c1
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T14.51-cafb833/job.log
 (1/2) openssh-tests.py:OpenSSH.test;run-run_type-upstream-7115:  FAIL: openssh-tests.py: 'make check' failed. (231.27 s)
 (2/2) openssh-tests.py:OpenSSH.test;run-run_type-distro-f473:  ERROR: openssh source install failed. (3.81 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T14.51-cafb833/results.html
JOB TIME   : 257.28 s



RHEL:
JOB ID     : c2e3b9177f68d1504a24de69107741b4061d84d6
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T04.23-c2e3b91/job.log
 (1/2) openssh-tests.py:OpenSSH.test;run-run_type-upstream-7115:  FAIL: openssh-tests.py: 'make check' failed. (461.93 s)
 (2/2) openssh-tests.py:OpenSSH.test;run-run_type-distro-f473:  FAIL: openssh-tests.py: 'make check' failed. (102.70 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T04.23-c2e3b91/results.html
JOB TIME   : 577.44 s
```

File: security/openssh-tests.py [hunk 1]
```diff
@@ -42,22 +42,40 @@ def setUp(self):
...
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/openssh/openssh-portable/archive/master.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'openssh-portable-master')
-        os.chdir(self.sourcedir)
-        process.run('autoreconf', ignore_status=True)
-        process.run('./configure --with-pam --with-libedit --with-kerberos5 \
-                    --with-selinux --with-md5-passwords', ignore_status=True)
+        run_type = self.params.get('type', default='upstream')
```
Comment (Naresh-ibm) [inline security/openssh-tests.py:45]: @nasastry make default='distro'

File: security/openssh-tests.py [hunk 2]
```diff
@@ -42,22 +42,40 @@ def setUp(self):
...
+        if run_type == "upstream":
+            def_url = ("https://github.com/openssh/openssh-portable/"
+                       "archive/master.zip")
+            url = self.params.get('url', default=def_url)
```
Comment (Naresh-ibm) [inline security/openssh-tests.py:49]: remove def_url variable and change default=None

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by Naresh-ibm — APPROVED** [review]
LGTM

**Pull request merged**